packages/smooth_app/lib/pages/scan/ml_kit_scan_page.dart

```diff
@@ -158,21 +145,59 @@ class MLKitScannerPageState extends State<MLKitScannerPage> {
         debugPrint(e.toString());
       }
     }
+
     if (mounted) {
       setState(() {});
     }
   }
 
+  void cameraListener() {
+    if (mounted) {
+      setState(() {});
+
+      if (_controller?.value.hasError == true) {
+        // TODO(M123): Handle errors better
+        debugPrint(_controller!.value.errorDescription);
+      }
+    }
+  }
+
   Future<void> _stopImageStream() async {
+    if (stoppingCamera) {
+      return;
+    }
+
     stoppingCamera = true;
     if (mounted) {
       setState(() {});
     }
+
+    _controller?.removeListener(cameraListener);
     await _controller?.dispose();
-    barcodeScanner?.close();
+    await barcodeScanner?.close();
+
     barcodeScanner = null;
-    _controller?.removeListener(() {});
     _controller = null;
+
+    if (mounted && ScreenVisibilityDetector.visible(context)) {
+      final DateTime referentialTime = DateTime.now();
+
+      WidgetsBinding.instance?.addPostFrameCallback((_) {
+        final int diff =
+            DateTime.now().difference(referentialTime).inMilliseconds;
+
+        // The screen is still visible, we should restart the camera
+        if (diff < 60) {
```

Comment:
Basically, my computation is as the following.

The average refresh rate of a screen is 60 frames per second, which means that every 16 milliseconds, we should be able to redraw what's on the screen. We have more and more devices with a higher refresh rate (some iPads and many Android devices).

But on the other hand, low-entry devices should skip some frames because their CPUs are less performant.
During my tests, 1 frame (16 milliseconds) was not always enough. Following an "empirical method", 60 (which does mean less than 4 frames) covered all my tests.

Reply:
What happens if some old devices take more than 60ms? Does it crash? Wouldn't we be safer with 100ms, without downgrading the user experience?
A constant, something like `static const double FIRST_FRAME_THRESHOLD_IN_MILLIS = 100;`, would make sense.
So would an additional comment like `// Through empirical method on 2021 devices 60ms seemed enough, but 100ms is safer for older devices`.

Reply:
If it takes more than 60 ms, it will be the same as today => a black screen.

I add some comments + the constant in my next commit